virtual-server/examples/python/vsclient.py

```diff
@@ -0,0 +1,128 @@
+        name = manifest['metadata']['name']
+        if not namespace or not name:
+            raise TypeError('VirtualServer metadata.namespace and metadata.name is required')
+        return self.api.create_namespaced_custom_object(
+            VSClient.GROUP, VSClient.VERSION, namespace, VSClient.PLURAL, manifest)
+
+    def update(self, manifest):
+        """
+        Update virtual server according to the manifest.
+        """
+        if not manifest['metadata']:
+            raise TypeError('VirtualServer metadata is required')
+        namespace = manifest['metadata']['namespace']
+        name = manifest['metadata']['name']
+        if not namespace or not name:
+            raise TypeError('VirtualServer metadata.namespace and metadata.name is required')
+        return self.api.patch_namespaced_custom_object(
+            VSClient.GROUP, VSClient.VERSION, namespace, VSClient.PLURAL, manifest)
+
+    @staticmethod
+    def match_condition(condition, expected_status):
+        if expected_status in VSClient.EXPECTED_CONDITIONS:
+            cond = VSClient.EXPECTED_CONDITIONS[expected_status]
+            if 'reason' in condition and cond['reason'] == condition['reason'] and \
+                    'status' in condition and cond['status'] == condition['status'] and \
+                    'type' in condition and cond['type'] == condition['type']:
+                return expected_status
+        return None
+
+    def ready(self, namespace, name, expected_state='Ready'):
+        """
+        Wait for virtual server until matches one of the expected condtion from EXPECTED_CONDITIONS
+        """
+        w = watch.Watch()
+        kwargs = {
+            'watch': True,
+            'field_selector': f'metadata.name={name}'
+        }
+        ready_condition = None
+        print('Waiting for virtual server to be ready ...')
+        for event in w.stream(self.api.list_namespaced_custom_object,
+                              VSClient.GROUP, VSClient.VERSION, namespace, VSClient.PLURAL, **kwargs):
+            status = event['object']['status']
+
+            # the virtual server was already deleted
+            if event['type'] == 'DELETE':
+                ready_condition = 'Deleted'
+                w.stop()
+                break
+
+            if 'conditions' not in status:
+                continue
+
+            # Check if new state matches the expected condition
+            ready_condition = VSClient.match_condition(status['conditions'][0], expected_state)
+
+            # Wait for next condition
+            if not ready_condition:
+                continue
+            elif ready_condition == 'Ready':
+                network = status['network']
+                tcp_ip = network['externalIP'] if 'externalIP' in network else ''
+                udp_ip = network['internalIP'] if 'internalIP' in network else ''
+                print(f'{name}, external IP: {tcp_ip}, internal IP: {udp_ip}')
```

Comment:
Actually, let's first change the variable names here from tcp_ip = .. and udp_ip = ... to externalIP and internalIP, as the original variable names are no longer relevant. Functionality-wise this is fine, just not clear.